src/Native/Unix/System.Native/pal_console.c

```diff
@@ -445,7 +445,7 @@ static bool InitializeTerminalCore()
 {
     g_haveInitTermios = tcgetattr(STDIN_FILENO, &g_initTermios) >= 0;
 
-    if (!g_haveInitTermios && (errno == ENOTTY || errno == ENODEV))
+    if (!g_haveInitTermios)
```

Comment:
We can get rid of `g_haveInitTermios` (`g_haveInitTermios == !g_noTty`).

Reply:
Maybe also turn g_noTty into g_hasTty.

Reply:
updated.